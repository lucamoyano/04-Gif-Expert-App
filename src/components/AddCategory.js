import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({ target: { value } }) => {
        setInputValue( value );
    }

    const handleSubmit = (e) => {
        e.preventDefault(); //Prevenir que se refresque la pagina
        
        if( inputValue.trim().length > 2 ) {
            setCategories( c => [inputValue, ...c]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text" 
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}