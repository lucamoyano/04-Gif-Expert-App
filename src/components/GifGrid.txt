import React from 'react'
import {GifGridItem} from './GifGridItem';
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ( {category} ) => {

    //const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs( category )
    //         //.then( imgs => setImages( imgs ) ); 
    //         //Como es una funcion que envia su primer argumento como primer argumento a la funcion
    //         //Se puede utilizar de la siguiente manera:
    //         .then( setImages );
    // }, [category]); 
    // //Le pasamod un arreglo vacio como segundo argumento
    // //Para decirle que se ejecute una sola vez

    const { data, loading } = useFetchGifs();

    return (
        <>
        <h3>{category}</h3>
        <div className="card-grid">
            {/* {
                images.map( ({id, title}) => (
                        <li key={id}> {title} </li>
                    )
                )
            } */}

            {/* {
                images.map( img => (
                    // <GifGridItem img={ img } />
                    <GifGridItem 
                        key={img.id}
                        { ...img }  //Le paso las props mediante desetructuracion
                    />
                    )
                )
            } */}
        </div>
        </>
    )
}
