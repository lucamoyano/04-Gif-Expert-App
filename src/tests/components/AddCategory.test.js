import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";


describe('Prueba en componente AddCategory', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories} /> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories} /> );
    });

    test('debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.props().onChange({target: {value} });
        //input.simulate('change',{target: {value} }); //tambien se puede hacer con simulate
        expect( wrapper.find('p').text().trim() ).toBe(value)
    });

    test('NO debe de poster la informacion onSubmit', () => {
        //wrapper.find('form').simulate('submit', {preventDefault()})
        wrapper.find('form').props().onSubmit({ preventDefault(){} })

        //No se haya llamado (not) 
        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('debe llamar el setCategories y limpiar la caja del texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        //1. Simular el input change
        input.props().onChange({target: {value} });

        //2. Simular el submit
        wrapper.find('form').props().onSubmit({ preventDefault(){} });

        //3. setCtegories debe ser llamado 
        //y que sea una funciÃ³n. setCategories( c => [inputValue, ...c]);
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );

        //4. el valor del input debe ser ''
        expect( input.prop('value') ).toBe('');
    })
    
    
})