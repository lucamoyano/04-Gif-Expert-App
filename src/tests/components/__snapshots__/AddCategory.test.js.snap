// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en componente AddCategory debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
